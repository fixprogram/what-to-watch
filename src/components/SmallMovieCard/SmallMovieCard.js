import React from 'react';
import PropTypes from 'prop-types';

const SmallMovieCard = (props) => {
  const {id, picture, title, onHover, outHover} = props;
  return (
    <article className="small-movie-card catalog__movies-card"
      onMouseOver={onHover}
      onMouseOut={outHover}
      data-key={id}>
      <div className="small-movie-card__image">
        <img src={picture} alt={title} width="280" height="175" />
      </div>
      <h3 className="small-movie-card__title">
        <a className="small-movie-card__link" href="movie-page.html">{title}</a>
      </h3>
    </article>
  );
};

SmallMovieCard.defaultProps = {
  onHover() {},
  outHover() {}
};

SmallMovieCard.propTypes = {
  id: PropTypes.string.isRequired,
  picture: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  onHover: PropTypes.func,
  outHover: PropTypes.func
};

export default SmallMovieCard;
