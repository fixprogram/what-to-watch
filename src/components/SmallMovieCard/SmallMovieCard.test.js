import React from 'react';
import renderer from 'react-test-renderer';
import SmallMovieCard from './SmallMovieCard';

it(`When user artist song is not sent`, () => {
  const tree = renderer.create(
      <SmallMovieCard
        id={`id1`}
        picture={`path`}
        title={`Movie`}
        onHover={jest.fn()}
        outHover={jest.fn()} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
}
);
