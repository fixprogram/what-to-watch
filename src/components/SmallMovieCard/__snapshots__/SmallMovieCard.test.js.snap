// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When user artist song is not sent 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  data-key="id1"
  onMouseOut={[MockFunction]}
  onMouseOver={[MockFunction]}
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Movie"
      height="175"
      src="path"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Movie
    </a>
  </h3>
</article>
`;
